internal/cli/root/builder.go

```diff
@@ -310,6 +317,11 @@ func shouldCheckCredentials(cmd *cobra.Command) AuthRequirements {
 			return r
 		}
 	}
+
+	if sourceType, ok := cmd.Annotations["sourceType"]; ok && sourceType == plugin.SourceType {
```

Comment:
[optional recommendation] move this logic to a plugin.IsPluginCmd(cmd *cobra.Command).
This way the entire logic (annotate/check annotation) is encapsulated